### PR TITLE
Use ArrayAccess interface instead of Doctrine Collection

### LINKED ISSUE #7538: Nested CollectionTypes
I have a similar problem to #6777. In the latest version 3.105.3 in AdminHelper, the line

`$childFormBuilder = $this->getChildFormBuilder($formBuilder, $elementId);`

in is returning null when I am using nested `Sonata\Form\Type\CollectionType`.

In my Entity I have to expose the Collection

```
public function getAnswers(): Collection
{
    return $this->answers;
}
```

instead of only the array

```
/**
 * @return array<Answer>
 */
public function getAnswers(): array
{
    return $this->answers->toArray();
}
```

Otherwise it will not work because the else block is reached in the AdminHelper class.

In the loop of `getChildFormBuilder` the value for `$elementId` is `s616839720b0e2_questions_0_answers`. But there is only a `s616839720b0e2_questions` in the array (iterator). Not sure if this is the case of the problem.

### PR DESCRIPTION
**Pull request by TheFox**
## Subject

I am targeting this branch, because it's a fix for version 3.

Closes #7538.

## Changelog

```markdown
### Changed

- Instead of using the Collection interface an easy switch to ArrayAccess interface can be used.
```

### CODE REVIEW

File: src/Admin/AdminHelper.php [hunk 1]
```diff
@@ -240,20 +240,12 @@ public function appendFormFieldElement(AdminInterface $admin, $subject, $element
 
             $collection = $this->propertyAccessor->getValue($subject, $path);
 
-            if (!($collection instanceof Collection)) {
```
Comment (jordisala1991) [inline src/Admin/AdminHelper.php:243]: What if it is not \ArrayAccess or native array?

Comment (TheFox) [reply src/Admin/AdminHelper.php:243]: Yes, you right. We need a check if it's \ArrayAccess or array.

**Review by VincentLanglet — COMMENTED** [review]
Does a unit test would be possible ?

File: src/Admin/AdminHelper.php [hunk 2]
```diff
@@ -240,10 +240,10 @@ public function appendFormFieldElement(AdminInterface $admin, $subject, $element
...
+            if (!($collection instanceof \ArrayAccess) && !is_array($collection)) {
```
Comment (jordisala1991) [inline src/Admin/AdminHelper.php:243]: I think the () around instanceof are unnecesary.

**Comment by TheFox** [comment]
I just realized that this would not fix the other issue in #7538. It's also about Nested Collections. When I add a nested collection in level 2 I get this error:

```
Could not get element id from s61695531a2b2e_questions_1_answers Failing part: answers
```

**Comment by VincentLanglet** [comment]
Hi, can you rebase ?

**Comment by jordisala1991** [comment]
I found one case where we use this strategy too, so this Pr makes sense to me. (the case is located at the appenParentObject of the on AbstractAdmin)

**Comment by VincentLanglet** [comment]
Do you have the time to finish the PR @TheFox ? :)

**Pull request closed without merging**